File: c#/vs/Common7/IDE/ProjectTemplates/CSharp/Silverlight/1042/BusinessApplication/Helpers/TargetNullValueConverter.cs
﻿namespace $safeprojectname$
{
    using System;
    using System.Windows.Data;

    /// <summary>
    /// 빈 string 값을 가질 수 있는 바인딩 가능한 개체에 대한 속성을 빈 문자열 값을 null로 설정해야 하는 종속성 속성에 바인딩할 수 있도록 해주는 양방향 IValueConverter입니다.
    /// </summary>
    public class TargetNullValueConverter : IValueConverter
    {
        /// <summary>
        /// <c>null</c> 또는 빈 strings를 <c>null</c>로 변환합니다.
        /// </summary>
        /// <param name="value">변환할 값입니다.</param>
        /// <param name="targetType">결과의 예상 형식입니다(무시됨).</param>
        /// <param name="parameter">선택적 매개 변수입니다(무시됨).</param>
        /// <param name="culture">변환의 문화권입니다(무시됨).</param>
        /// <returns><paramref name="value"/>가 <c>null</c>이거나 비어 있으면 이 메서드는 <c>null</c>을 반환하고, 그렇지 않으면 <paramref name="value"/>를 반환합니다.</returns>
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            string strValue = value as string;

            return string.IsNullOrEmpty(strValue) ? null : value;
        }

        /// <summary>
        /// <c>null</c>을 <see cref="String.Empty"/>로 다시 변환합니다.
        /// </summary>
        /// <param name="value">변환할 값입니다.</param>
        /// <param name="targetType">결과의 예상 형식입니다(무시됨).</param>
        /// <param name="parameter">선택적 매개 변수입니다(무시됨).</param>
        /// <param name="culture">변환의 문화권입니다(무시됨).</param>
        /// <returns><paramref name="value"/>가 <c>null</c>이면 <see cref="String.Empty"/>를 반환하고, 그렇지 않으면 <paramref name="value"/>를 반환합니다.</returns>
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return value ?? string.Empty;
        }
    }
}

File: c#/vs/Common7/IDE/ProjectTemplatesCache/VisualBasic/Silverlight/1028/BusinessApplication/Helpers/TargetNullValueConverter.vb
﻿Imports System
Imports System.Windows.Data

''' <summary>
''' 雙向 IValueConverter，可以讓您將可為空 string 值的可繫結物件上的屬性繫結到在該情況下應設定為 null 的相依性屬性。
''' </summary>
Public Class TargetNullValueConverter
    Implements IValueConverter

    ''' <summary>
    ''' 將 <c>null</c> 或空 strings 轉換成 <c>null</c>。
    ''' </summary>
    ''' <param name="value">要轉換的值。</param>
    ''' <param name="targetType">此結果的預期型別 (忽略)。</param>
    ''' <param name="parameter">選擇性參數 (忽略)。</param>
    ''' <param name="culture">此轉換的文化特性 (忽略)。</param>
    ''' <returns>如果 <paramref name="value"/> 為 <c>null</c> 或空白，此方法會傳回 <c>null</c>，否則它將傳回 <paramref name="value"/>。</returns>
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Dim strValue As String = Nothing

        If TypeOf value Is String Then
            strValue = DirectCast(value, String)
        End If

        Return If(String.IsNullOrEmpty(strValue), Nothing, value)
    End Function

    ''' <summary>
    ''' 將 <c>null</c> 轉換回 <see cref="String.Empty"/>。
    ''' </summary>
    ''' <param name="value">要轉換的值。</param>
    ''' <param name="targetType">此結果的預期 type (忽略)。</param>
    ''' <param name="parameter">選擇性參數 (忽略)。</param>
    ''' <param name="culture">此轉換的文化特性 (忽略)。</param>
    ''' <returns>如果 <paramref name="value"/> 為 <c>null</c>，它會傳回 <see cref="String.Empty"/>，否則將傳回 <paramref name="value"/>。</returns>
    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return If(value Is Nothing, String.Empty, value)
    End Function
End Class